include/hipSYCL/sycl/libkernel/vec.hpp

```diff
@@ -154,44 +154,42 @@ struct logical_vector_op_result
 template<> struct logical_vector_op_result<char>
 { using type = int8_t; };
 
-template<> struct logical_vector_op_result<int8_t>
+template<> struct logical_vector_op_result<signed char>
 { using type = int8_t; };
 
-template<> struct logical_vector_op_result<uint8_t>
+template<> struct logical_vector_op_result<unsigned char>
 { using type = int8_t; };
 
-template<> struct logical_vector_op_result<int16_t>
+template<> struct logical_vector_op_result<short>
 { using type = int16_t; };
 
-template<> struct logical_vector_op_result<uint16_t>
+template<> struct logical_vector_op_result<unsigned short>
 { using type = int16_t; };
 
 template<> struct logical_vector_op_result<float>
 { using type = int32_t; };
 
-template<> struct logical_vector_op_result<uint32_t>
+template<> struct logical_vector_op_result<unsigned>
 { using type = int32_t; };
 
-template<> struct logical_vector_op_result<int32_t>
+template<> struct logical_vector_op_result<int>
+{ using type = int32_t; };
+
+template<> struct logical_vector_op_result<long>
```

Comment:
Ah, thanks, it should return int64_t on Linux. We can just use `sizeof(long)` :) In practice there won't really be truncation because this is only used for the return type of boolean operators, so will store 0 or 1.

Reply:
It should now automagically do the correct thing for any type, and do so with less code :)